lib/types.ts

```diff
@@ -0,0 +1,25 @@
+export interface State {
+  cached?: boolean;
+  checkOnly?: boolean;
+}
+
+export interface Webhook<TExample extends object = object> {
+  name: string;
+  description: string;
+  actions: string[];
+  examples: TExample[];
+}
+
+export interface Section {
+  id: string;
+  html: string;
+}
+
+// polyfills for types used by turndown to avoid having the include the DOM lib
+// as they'd be considered globally available in the whole codebase by typescript
+// (remove if we actually start using DOM elements e.g. via jsdom)
+declare global {
```

Comment:
Could you explain the reasoning here? Why wouldn't we just include the DOM lib, and call it a day

Reply:
I try to avoid including the DOM lib on non-dom projects because you can't scope their availability to specific files easily - so it becomes "valid" to use them anywhere, and autocompletion in IDEs will start drawing from those types which can be confusing.

It's also not easy to tell when reviewing if a type is coming from the DOM, e.g. `Element` is a type in the DOM lib, meaning you could reference it by mistake when you really want `cheerio.Element`. (a few other good ones include `Url` , `Blob`, and `Timer` - that's an especially annoying one as `NodeJS.Timer` doesn't have the same type as `DOM.Timer`)

Hence for situations like this (where we're just doing `turnabout.turnabout`, and so not interacting with the section that uses the DOM libs) I like to favor declaring empty interfaces so I don't have to worry about it.